### ISSUE TITLE
SA1600 triggers for private/internal interface members

### ISSUE THREAD

**Issue by gtbuchanan** [DotNetAnalyzers/StyleCopAnalyzers#3551]
[Interfaces may have private and internal members](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/types/interfaces):

> Interface members are public by default, and you can explicitly specify accessibility modifiers, such as public, protected, internal, private, protected internal, or private protected. A private member must have a default implementation.

SA1600 requires documentation of these members when `documentPrivateElements: false` and `documentInternalElements: false`. For example, `InternalMethod` and `PrivateMethod` here are triggering SA1600:

```csharp
/// <summary>
/// An interface with private members.
/// </summary>
internal interface IInterfaceWithPrivateMembers
{
    internal void InternalMethod()
    {
    }

    private void PrivateMethod()
    {
    }

    /// <summary>
    /// A public method.
    /// </summary>
    public void PublicMethod() => this.PrivateMethod();
}
```

Tested with 1.2.0-beta.435.

**Comment by bjornhellander** [comment]
See the description of `documentInterfaces `in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md?plain=1#L577:
_Specifies whether interface members need to be documented. When true, all interface members require documentation, regardless of accessibility._
Seems to be why this is happening.

**Comment by bjornhellander** [comment]
Personally, I think the behaviour for internal members with bodies are still correct, since implementors of the interface still needs to understand what the method is supposed to do to be able to override it when necesssary.

Private methods with bodies is a different matter, since they can not be overridden or used by anyone else. I would find it reasonable if `documentInterfaces` did not apply to private members.

**Comment by gtbuchanan** [comment]
@bjornhellander What you're saying about internal interface members definitely makes sense in regards to `documentInterfaces`. I will update the title/description to only mention private members. I agree that private members should probably be treated as a special case.

**Comment by bjornhellander** [comment]
Any thoughts on @gtbuchanan's suggestion, @sharwell?
